docs/developer/logging-guidelines.md

```diff
@@ -0,0 +1,407 @@
+  (example - eventHandler)
+- *Relevant extra context* - additional context depending on the location of the log line in the code.
+
+Additionally, specifically for the developer persona:
+
+- *Stacktrace* - to pinpoint where in the code the message was logged.
+
+The next section shows how to add context.
+
+#### Examples of Log Messages
+
+1. A simple info message with a timestamp with the default `info` level:
+
+   ```go
+   logger.Info("Info message")
+   ```
+
+   ```json
+   {"level":"info","ts":"2023-07-20T15:10:03-04:00","msg":"Info message"}
+   ```
+
+2. An info message with the `debug` level:
+
+   ```go
+    logger.V(1).Info("Info message")
+   ```
+
+   ```json
+   {"level":"debug","ts":"2023-07-20T15:10:03-04:00","msg":"Info message"}
+   ```
+
+3. An error:
+
+   ```go
+    logger.Error(errors.New("test error"), "Error message")
+   ```
+
+   ```json
+    {"level":"error","ts":"2023-07-20T15:10:03-04:00","msg":"Error message","error":"test error","stacktrace":"main.main\n\t/<REDACTED for the document>.go:73\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
+   ```
+
+   > An error message includes the stack trace by default.
+4. A message with a Kubernetes resource:
+
+   ```go
+   // hr implements client.Object (controller-runtime)
+   // hr is *v1beta1.HTTPRoute
+    logger.Info("Orocessed resource",
+        "resource", hr)
+   ```
+
+   ```json
+   {"level":"info","ts":"2023-07-20T15:10:03-04:00","msg":"Processed resource","resource":{"apiVersion":"gateway.networking.k8s.io/v1beta1","kind":"HTTPRoute","namespace":"test","name":"hr-1"}}
+   ```
+
+   > The resource must include `TypeMeta`, otherwise its `apiVersion` and `kind` will not be printed.
+5. A message with a component name:
+
+   ```go
+    logger = logger.WithName("component-a")
+    logger.Info("Hello")
+   ```
+
+   ```json
+    {"level":"info","ts":"2023-07-20T15:10:03-04:00","logger":"component-a","msg":"Hello"}
+   ```
+
+6. A message with extra context:
+
+   ```go
+    logger.Info("info message",
+        "status", "ok",
+        "generation", 123)
+   ```
+
+   ```json
+   {"level":"info","ts":"2023-07-20T15:10:03-04:00","msg":"Info message","status":"ok","generation":123}
+   ```
+
+7. An info message with a stack trace configured for the `info` level:
+
+   ```go
+    logger.Info("Info message")
+   ```
+
+   ```json
+   {"level":"info","ts":"2023-07-20T15:10:03-04:00","msg":"Info message","stacktrace":"main.main\n\t/<REDACTED for the document>.go:100\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
+   ```
+
+#### Message Guidelines
+
+1. Start a message with an uppercase letter. For example, `"Reloaded nginx"`.
+2. Do not put dynamic values in the message, add them as key values (extra context):
+
+   DO NOT:
+
+   ```go
+    logger.Info(fmt.Sprintf("Got response with status %q", "ok"))
+   ```
+
+   DO:
+
+   ```go
+    logger.Info("Got response",
+        "status", "ok")
```

Comment:
```suggestion
        "status", "ok",
    )
```

To be consistent with our go style guide. Same with the other multi-line examples.